docs: correct example for RouteReuseStrategy getRouteKey

SkyZeroZx opened the pull request:
Update the example to properly demonstrate the getRouteKey method usage fixes #65021

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number:  https://github.com/angular/angular/issues/65021


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

SkyZeroZx commented on adev/src/content/guide/routing/customizing-route-behavior.md:
```diff
@@ -198,7 +198,12 @@ export class CustomRouteReuseStrategy implements RouteReuseStrategy {
   }
 
   private getRouteKey(route: ActivatedRouteSnapshot): string {
```
It might work, but if I'm not mistaken, we could have path collisions when a subroute has the same path.

atscott replied:
You cannot effectively key off of path. Paths may be duplicated with canMatch guards.

SkyZeroZx committed:
docs: correct example for RouteReuseStrategy getRouteKey

Update the example to properly demonstrate the getRouteKey method usage fixes #65021
